fx-src/FirefoxStorage.js

```diff
@@ -299,6 +302,40 @@ export default class FirefoxAdapter extends BaseAdapter {
         return result[0].getResultByName("last_modified");
       });
   }
+
+  /**
+   * Reset the sync status of every record and collection we have
+   * access to.
+   */
+  resetSyncStatus() {
+    // We're going to use execute instead of executeCached, so build
+    // in our own sanity check
+    if (!this._connection) {
+      throw new Error("The storage adapter is not open");
+    }
+
+    return this._connection.executeTransaction(function* (conn) {
+      const promises = [];
+      yield conn.execute(statements.scanAllRecords, null, function(row) {
+        const record = JSON.parse(row.getResultByName("record"));
+        const record_id = row.getResultByName("record_id");
+        const collection_name = row.getResultByName("collection_name");
+        if (record._status === "deleted") {
+          // Garbage collect deleted records.
+          promises.push(conn.execute(statements.deleteData, {collection_name, record_id}));
+        }
+        else {
+          const newRecord = Object.assign({}, record, {
+            _status: "created",
+            last_modified: undefined,
+          });
```

Comment:
nitpick: we use object rest spread (maybe you left it like that 'cuz you know it'll be moved to gecko)